tests/integration/testAuthClient.js

```diff
@@ -0,0 +1,394 @@
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+var test = require('tape');
+var nock = require('nock');
+var AuthClient = require('../../lib/google-apis/auth-client.js');
+var ErrorMessage = require('../../lib/classes/error-message.js');
+var lodash = require('lodash');
+var isObject = lodash.isObject;
+var isString = lodash.isString;
+var isEmpty = lodash.isEmpty;
+var has = lodash.has;
+var client;
+
+test(
+  'Test given valid init parameters we should be able to create a client',
+  function ( t ) {
+
+    if (!isString(process.env.GCLOUD_PROJECT)) {
+      t.fail("The gcloud project id (GCLOUD_PROJECT) was not set as an env variable");
+      t.end();
+      process.exit();
+    }
+
+    try {
+      client = new AuthClient(process.env.GCLOUD_PROJECT, true);
```

Comment:
It looks like `tape` doesn't have test lifecycle hooks the way mocha does (https://github.com/substack/tape/issues/59). These tests can be restructured to be order independent (setting the auth client in each test). I don't have a strong preference between test frameworks as we have uniformity between the agents.